### PR TITLE
[Doc] Reimplement Azure Databricks deployment guide to use VNet injection & NPIP

### LINKED ISSUE #2093: [DOC] Reimplement Azure Databricks deployment guide to use VNet injection & NPIP
Right now, [Azure Databricks deployment guide](https://registry.terraform.io/providers/databricks/databricks/latest/docs/guides/azure-workspace) is showing how to create ADB workspace, but it's relying on the default settings that doesn't follow the best practices regarding VNet injection & NoPublicIP (SCC). It would be better to reimplement this to show the use of the best practices.

### PR DESCRIPTION
**Pull request by dgomez04**
## Changes

Updated the Azure Databricks deployment guide to implement VNet injection and secure cluster connectivity, following best practices. Ran `make fmt-docs` and tested the deployment myself. 

Closes #2093 
## Tests

- [ ] `make test` run locally
- [x] relevant change in `docs/` folder
- [ ] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [ ] using Go SDK

### CODE REVIEW

File: docs/guides/azure-workspace.md [hunk 1]
```diff
@@ -54,13 +65,89 @@ resource "azurerm_resource_group" "this" {
...
+    virtual_network_id                                   = azurerm_virtual_network.this.id
+    private_subnet_name                                  = azurerm_subnet.private.name
+    public_subnet_name                                   = azurerm_subnet.public.name
+    public_subnet_network_security_group_association_id  = azurerm_subnet_network_security_group_association.public.id
+    private_subnet_network_security_group_association_id = azurerm_subnet_network_security_group_association.private.id
+  }
+  depends_on = [
+    azurerm_subnet_network_security_group_association.public,
+    azurerm_subnet_network_security_group_association.private
+  ]
```
Comment (alexott) [inline docs/guides/azure-workspace.md:150]: We don't need explicit `depends_on` here as we have explicit dependencies on lines 144-145

File: docs/resources/dashboard.md
Comment (alexott) [inline docs/resources/dashboard.md:1]: let make this & other formatting changes in a separate PR...

**Review by alexott — CHANGES_REQUESTED** [review]
small changes required, otherwise it's good

File: docs/guides/azure-workspace.md (hunk 1 above)
Comment (dgomez04) [reply docs/guides/azure-workspace.md]: Done ✔️

File: docs/resources/dashboard.md
Comment (dgomez04) [reply docs/resources/dashboard.md:1]: Done ✔️ I've returned them to their original state from `main` branch. Let me know if that works.

**Review by alexott — APPROVED** [review]

**Pull request merged**